bigtable/admin.go

```diff
@@ -432,6 +474,33 @@ func (ac *AdminClient) updateTableWithConf(ctx context.Context, conf *UpdateTabl
 		}
 	}
 
+	if conf.automatedBackupPolicy != nil {
+		// Update Automated Backup Policy
+		if *conf.automatedBackupPolicy == DisableAutomatedBackupPolicy {
+			// Disable Automated Backup Policy
+			updateMask.Paths = append(updateMask.Paths, "automated_backup_policy")
+		} else if conf.automatedBackupPolicy.Frequency != nil && conf.automatedBackupPolicy.RetentionPeriod != nil {
+			// Update both Retention Period and Frequency
+			updateMask.Paths = append(updateMask.Paths, "automated_backup_policy")
+			frequency := durationpb.New(conf.automatedBackupPolicy.Frequency.(time.Duration))
+			retentionPeriod := durationpb.New(conf.automatedBackupPolicy.RetentionPeriod.(time.Duration))
+			automated_backup_policy := &btapb.Table_AutomatedBackupPolicy{RetentionPeriod: retentionPeriod, Frequency: frequency}
+			req.Table.AutomatedBackupConfig = &btapb.Table_AutomatedBackupPolicy_{AutomatedBackupPolicy: automated_backup_policy}
+		} else if conf.automatedBackupPolicy.Frequency != nil {
+			// Update Frequency
+			updateMask.Paths = append(updateMask.Paths, "automated_backup_policy.frequency")
+			frequency := durationpb.New(conf.automatedBackupPolicy.Frequency.(time.Duration))
+			automated_backup_policy := &btapb.Table_AutomatedBackupPolicy{Frequency: frequency}
+			req.Table.AutomatedBackupConfig = &btapb.Table_AutomatedBackupPolicy_{AutomatedBackupPolicy: automated_backup_policy}
+		} else if conf.automatedBackupPolicy.RetentionPeriod != nil {
+			// Update Retention Period
+			updateMask.Paths = append(updateMask.Paths, "automated_backup_policy.retention_period")
+			retentionPeriod := durationpb.New(conf.automatedBackupPolicy.RetentionPeriod.(time.Duration))
+			automated_backup_policy := &btapb.Table_AutomatedBackupPolicy{RetentionPeriod: retentionPeriod}
+			req.Table.AutomatedBackupConfig = &btapb.Table_AutomatedBackupPolicy_{AutomatedBackupPolicy: automated_backup_policy}
+		}
```

Comment:
Is there a difference between a field mask that contains a single path `automated_backup_policy` and a field mask that contains 2 paths: `automated_backup_policy.frequency` and `automated_backup_policy.retention`?

If not, I think a nested if would be better:

if conf.automatedBackupPolicy != nil {
  if conf.automatedBackupPolicy.Frequency != nil {
    // set the freq & update automated_backup_policy.frequency
  }
  if conf.automatedBackupPolicy.Frequency != nil {
    //
  }
}

Reply:
Thanks, updated